### PR TITLE
modify layers reduction

### PR DESCRIPTION
**Pull request by Ikochoy**

### CODE REVIEW

File: Models/ImageEncoder.py [hunk 1]
```diff
@@ -25,19 +25,33 @@ def __init__(self, choice, dropout=False, fine_tune=False):
...
+
     elif choice == "AlexNet":
       self.model_name = "AlexNet"
       model_ft = models.alexnet(pretrained=True)
+      self.model = model_ft.to(self.device)  
+      # remove classifier layer
+      net = list(model_ft.children())[-1]
+      net_layers = list(net.children())[:-2]
+      net = nn.Sequential(*net_layers)
+      model_layers = list(model_ft.children())[:-1] + [net]
```
Comment (parinitaedke) [inline Models/ImageEncoder.py:43]: What is this doing exactly? I didn't understand.
It seems like net is the last FC layer, net layers if the model minus the last two layers, model_layers is just the full model again?

Comment (parinitaedke) [inline Models/ImageEncoder.py:41]: What is this doing exactly? I didn't understand.
It seems like net is the last FC layer, net layers if the model minus the last two layers, model_layers is just the full model again?

Comment (Ikochoy) [reply Models/ImageEncoder.py:43]: the net is the last sequential layers which refers to 6 children layers, and we want to remove the last 2 children layers from that sequential layer

File: Models/ImageEncoder.py [hunk 2]
```diff
@@ -25,19 +25,33 @@ def __init__(self, choice, dropout=False, fine_tune=False):
       # GoogLeNet
       self.model_name = "GoogLeNet_Inception_v3"
       model_ft = models.inception_v3(pretrained=True)
-     
+      self.model = model_ft.to(self.device)  
+      # remove the dropout + linear classifier layer
+      layers = list(self.model.to(self.device).children())[:-2]
+      self.model = nn.Sequential(*layers)
```
Comment (parinitaedke) [inline Models/ImageEncoder.py:33]: What is the output dims of this?

**Pull request merged**